bofire/data_models/features/categorical.py

```diff
@@ -359,38 +361,63 @@ class CategoricalOutput(Output):
     order_id: ClassVar[int] = 9
 
     categories: TCategoryVals
-    objective: Annotated[List[Annotated[float, Field(ge=0, le=1)]], Field(min_length=2)]
+    objective: Optional[AnyCategoricalObjective] = Field(
+        default_factory=lambda: ConstrainedCategoricalObjective(
```

Comment:
Should we set desireabilities to be an array of `True`'s in this instance?

Reply:
This actually ties into the above questions (i.e. regarding `None` types). I am also not sure of how to instantiate in this instance since it seems sort of "chicken-and-egg." Either way, we just need a default case here. Let me know what you prefer.

Reply:
Let us do the following, we implement the default generation into the field validator for the CategeforicalOutput:


``` python
def validate_objective(cls, objective, info):
    if objective is None:
         return ConstrainedCategoricalObjective(categories=info.data[`categories`], desirabilities = [True]*len(info.data[`categories`])]
    else:
        pass # do the validaton already implemented in the validator.
```

By this we are then setting as default that everything is allowed, which is the same as ignorig the CategoricalOutput in the optimization. 

In the attribute, we then set the folowing:

`objective: Optional[AnyCategoricalObjective] = Field(default=None, validate_default=True)`

Does this makes sense to you?

Reply:
Intuitively this makes sense, however, just applying this as is actually causes the following error: FAILED tests/bofire/data_models/serialization/test_deserialization.py::test_outputs_should_be_deserializable[outputs_spec0] 

This does not make sense since these output types are Continuous. Care to comment?